core/arch/x86_64/mm/core_mmu.c

```diff
@@ -622,6 +694,29 @@ static void add_va_space(struct tee_mmap_region *memory_map, size_t num_elems,
 	memory_map[n].size = size;
 }
 
+static bool core_mmu_place_tee_ram_at_top(paddr_t paddr)
+{
+	size_t l1size = 1ul << PML4_SHIFT;
+	paddr_t l1mask = l1size - 1;
+
+	return (paddr & l1mask) > (l1size / 2);
+}
+
+static void core_mmu_set_info_table(struct core_mmu_table_info *tbl_info,
+		unsigned int level, vaddr_t va_base, void *table)
+{
+	tbl_info->level = level;
+	tbl_info->table = table;
+	tbl_info->va_base = va_base;
+	tbl_info->shift = 0;
+	assert(level <= 3);
+
+	if (level == 1)
```

Comment:
Since num_entries always equal to 1,  if & else is not required. please double check.